src/registrar/tests/test_models_domain.py

```diff
@@ -1,54 +1,487 @@
+            Then a user-friendly error message is returned for displaying on the web
+        """
+        raise
+
+
+class TestAnalystLock(TestCase):
+    """Rule: Analysts may lock or unlock a domain to prevent or allow updates"""
+
+    def setUp(self):
+        """
+        Background:
+            Given the analyst is logged in
+            And a domain exists in the registry
+        """
+        pass
+
+    @skip("not implemented yet")
+    def test_analyst_locks_domain(self):
+        """
+        Scenario: Analyst locks a domain to prevent edits or deletion
+            When `domain.lock()` is called
+            Then `CLIENT_DELETE_PROHIBITED` is added to the domain's statuses
+            And `CLIENT_TRANSFER_PROHIBITED` is added to the domain's statuses
+            And `CLIENT_UPDATE_PROHIBITED` is added to the domain's statuses
+        """
+        raise
+
+    @skip("not implemented yet")
+    def test_analyst_locks_domain_idempotent(self):
+        """
+        Scenario: Analyst tries to lock a domain twice
+            Given `CLIENT_*_PROHIBITED` is already in the domain's statuses
+            When `domain.lock()` is called
+            Then Domain returns normally (without error)
+        """
+        raise
+
+    @skip("not implemented yet")
+    def test_analyst_removes_client_hold(self):
+        """
+        Scenario: Analyst unlocks a domain to allow deletion or edits
+            Given `CLIENT_*_PROHIBITED` is in the domain's statuses
+            When `domain.unlock()` is called
+            Then `CLIENT_DELETE_PROHIBITED` is no longer in the domain's statuses
+            And `CLIENT_TRANSFER_PROHIBITED` is no longer in the domain's statuses
+            And `CLIENT_UPDATE_PROHIBITED` is no longer in the domain's statuses
+        """
+        raise
+
+    @skip("not implemented yet")
+    def test_analyst_removes_client_hold_idempotent(self):
+        """
+        Scenario: Analyst tries to unlock a domain twice
+            Given `CLIENT_*_PROHIBITED` is not in the domain's statuses
+            When `domain.unlock()` is called
+            Then Domain returns normally (without error)
+        """
+        raise
+
+    @skip("not implemented yet")
+    def test_update_is_unsuccessful(self):
+        """
+        Scenario: An update to lock or unlock a domain is unsuccessful
+            When an error is returned from epplibwrapper
+            Then a user-friendly error message is returned for displaying on the web
+        """
+        raise
+
+
+class TestAnalystDelete(TestCase):
+    """Rule: Analysts may delete a domain"""
+
+    def setUp(self):
+        """
+        Background:
+            Given the analyst is logged in
+            And a domain exists in the registry
+        """
+        pass
+
+    @skip("not implemented yet")
+    def test_analyst_deletes_domain(self):
+        """
+        Scenario: Analyst permanently deletes a domain
+            When `domain.delete()` is called
```

Comment:
Just curious why do we just change the status to deleted and not remove it from the user and our tables? I assume it's so it can be brought back or to retain a record, but just curious.

Reply:
Basically that. To keep a record for future use and to help CISA meet their document retention policies.